include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,301 @@
+
+					auto finalPixelBuffer = core::make_smart_refctd_ptr<asset::ICPUBuffer>(state->mainRowScratchCacheByteSize);
+					auto finalPixel = reinterpret_cast<decodeType*>(finalPixelBuffer->getPointer());
+					memset(finalPixel, 0, finalPixelBuffer->getSize());
+
+					decodeType* decodeBuffer = scratchMemory + outDataAdressOffsetScratch + layersOffset;
+
+					for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+					{
+						*(currentColumnCache + channel) += *(mainRowCacheScratch + channel) += decodeBuffer[channel];
+						*(finalPixel + channel) = *(currentColumnCache + channel) - (ExclusiveMode ? decodeBuffer[channel] : 0);
+					}
+
+					memcpy(scratchMemory + outDataAdressOffsetScratch + layersOffset, finalPixel, state->mainRowScratchCacheByteSize);
+
+					if (newReadBlockPos.x == trueOutExtent.width - 1) // reset row cache when changing y
+					{
+						if (newReadBlockPos.y == 0) // sum values in (maxX, 0, z) and remove values from column cache
+						{ 
+							memcpy(totalImageCacheScratch + (newReadBlockPos.w * state->extent.depth + newReadBlockPos.z) * state->mainRowScratchCacheByteSize, finalPixel, state->mainRowScratchCacheByteSize);
+							memset(mainColumnCacheScratch, 0, state->mainColumnScratchCacheByteSize);
+						}
+
+						memset(mainRowCacheScratch, 0, state->mainRowScratchCacheByteSize);
+					}
+				};
+
+				{
+					core::vector3du32_SIMD localCoord;
+					for (auto& w = localCoord[3] = 0u; w < arrayLayers; ++w)
+						for (auto& z = localCoord[2] = 0u; z < trueOutExtent.depth; ++z)
+							for (auto& y = localCoord[1] = 0u; y < trueOutExtent.height; ++y)
+								for (auto& x = localCoord[0] = 0u; x < trueOutExtent.width; ++x)
+									sum(core::vectorSIMDu32( x, y, z, w ));
+				}
+
+				auto normalizeScratch = [&]()
+				{
+					core::vector3du32_SIMD localCoord;
+					for (auto& w = localCoord[3] = 0u; w < arrayLayers; ++w)
+						for (auto& z = localCoord[2] = 0u; z < trueOutExtent.depth; ++z)
+							for (auto& y = localCoord[1] = 0u; y < trueOutExtent.height; ++y)
+								for (auto& x = localCoord[0] = 0u; x < trueOutExtent.width; ++x)
+								{
+									const auto scratchInPlaneOffset = ((localCoord.z * state->extent.height + localCoord.y) * state->extent.width + localCoord.x);
+									const auto outDataAdressOffsetScratch = scratchInPlaneOffset * currentChannelCount;
+
+									auto* entryScratchAdress = scratchMemory + outDataAdressOffsetScratch + singleScratchLayerOffset * w;
+									auto* entryTotalImageCacheValues = reinterpret_cast<decodeType*>(totalImageCacheScratch + (localCoord.w * state->extent.depth + localCoord.z) * state->mainRowScratchCacheByteSize);
+
+									for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
+										entryScratchAdress[channel] /= entryTotalImageCacheValues[channel];
+								}
+				};
+
+				if (state->normalizeImageByTotalSATValues)
+					if (!asset::isNormalizedFormat(inFormat))
```

Comment:
erm if its a normalized format you definitely WANT TO normalize

Reply:
to fix

Reply:
update me when done.